packages/xo-web/src/xo-app/menu/index.js

```diff
@@ -25,26 +26,46 @@ import {
   getXoaState,
   isAdmin,
 } from 'selectors'
-import { every, identity, isEmpty, map } from 'lodash'
+import { every, flatten, identity, isEmpty, map, size, uniq } from 'lodash'
 
 import styles from './index.css'
 
 const returnTrue = () => true
+const missingPatchWarning = missingPatchCount =>
+  missingPatchCount > 0 ? (
+    <span>
+      <Tooltip content={_('homeMissingPatches')}>
+        <span className='text-warning'>
+          <Icon icon='alarm' />
+        </span>
+      </Tooltip>
+    </span>
+  ) : null
 
 @connectStore(
-  () => ({
-    isAdmin,
-    isPoolAdmin: getIsPoolAdmin,
-    nHosts: createGetObjectsOfType('host').count(),
-    nTasks: createGetObjectsOfType('task').count([
-      task => task.status === 'pending',
-    ]),
-    pools: createGetObjectsOfType('pool'),
-    srs: createGetObjectsOfType('SR'),
-    status: getStatus,
-    user: getUser,
-    xoaState: getXoaState,
-  }),
+  () => {
+    const getHosts = createGetObjectsOfType('host')
+    const getMissingPatches = createSelector(getHosts, hosts => {
+      return Promise.all(
+        map(hosts, host => getHostMissingPatches(host))
+      ).then(patches => uniq(flatten(patches)))
```

Comment:
Wouldn't a normal subscription like we usually do be better than a promise prop?